Week3/homework/project-card-validator.js

```diff
@@ -0,0 +1,51 @@
+// Checks the lenght of the string, returns true if it is 16.
+function checkLength(item){
+  return item.length === 16;
+}
+
+// Checks if the string consists of only digits.
+function checkAllDigit(item){
+  return +item == item;
+}
+
+// Checks if all the chars are same.
+function checkAllSame(item){
+  return item % 1111111111111111 === 0;
+}
+
+// Checks if the card number ends with an even number.
+function checkEven(item) {
+  return item % 2 === 0;
+}
+
+// Checks if the sum of all digits are more then 16.
+function checkSum(item){
+  return item.split('').map(function(a){
+    return +a
+  }).reduce(function(a, b){
+    return a + b;
+  }) > 16;
+}
+
+// It runs the functions above in a specific order to check if the card is valid.
+function validateCard(card){
+  if (checkLength(card)) {
+    if (checkAllDigit(card)) {
+      if (!checkAllSame(card)) {
+        if (checkEven(card)) {
+          if (checkSum(card)) {
+            return `${card} is VALID!`;
+          }
+        }
+      }
+    }
```

Comment:
👍 for splitting all logic in separate functions but these nested if statements aren't written in the best way. You can also think of using && operator and combine all of them in a single `if` statement. 🙂

Reply:
I can't say I like the callback hell like view here as well. But, I did not know if I should return a different fail message for each one. That is why I structured it this way. 
Thanks a lot. 
I'll check if I can create a better way to represent all possible outcomes.